src/quick-lint-js/lint.h

```diff
@@ -214,7 +216,8 @@ class linter {
 
   void report_error_if_assignment_is_illegal(
       const declared_variable *var, const identifier &assignment,
-      bool is_assigned_before_declaration) const;
+      bool is_assigned_before_declaration,
+      bool is_used_before_declaration = false) const;
```

Comment:
Minor: Make this parameter required, not optional. I've had too many mistakes with optional parameters (especially optional boolean parameters).